### ISSUE TITLE
Cannot reproduce the results shown in the paper? 

### ISSUE THREAD

**Issue by VoyageWang** [yihedeng9/STIC#1]
### Describe the issue

Hi there. A nice work! When I tried to reproduce the result of STIC, I did not see the improvement of STIC-stage1-preference optimization. The training setting is the same as yours. I tried two different versions of LLAVA 1.6 (vicuna-7b and mistral-7b) on the Science QA test dataset.  Here I report the results:

llava-mistral-7b on Science-QA test without STIC-stage1
![image](https://github.com/yihedeng9/STIC/assets/92557104/2e3f1d56-60b9-4108-b9af-9dbe710395be)

llava-mistral-7b on Science-QA test with STIC-stage1 (use your provided weight of lora)
![image](https://github.com/yihedeng9/STIC/assets/92557104/1bf08f33-2a29-45d2-8410-88d30fb80b83)

Here somehow, we saw the results but it is not consistent with the paper (approximately 60).

And I tried STIC-stage1 in the llava-vicuna-7b. Here I saw no improvement.  we did not change the trainingdate and setting.
llava-vicuna-7b (original)
![image](https://github.com/yihedeng9/STIC/assets/92557104/0690ecdd-62a4-46b8-a310-8b0060abfa46)
llava-vicuna-7b after STIC-stage1 (trained on 4 48G L20 in our envs)
![image](https://github.com/yihedeng9/STIC/assets/92557104/34ab288b-d8c2-4476-9e21-aea78bb5c561)

Here I also share the training loss log and lora setting here. They look normal

![image](https://github.com/yihedeng9/STIC/assets/92557104/a223e813-27cc-4d32-bf44-080f0a217710)

How should I do to get improvement in preference optimization? It really helps me. Thank you

**Comment by ZihaoZheng98** [comment]
> ### 描述问题
> 你好。干得不错！当我尝试重现 STIC 的结果时，我没有看到 STIC-stage1-preference 优化的改进。训练设置与你的相同。我在 Science QA 测试数据集上尝试了两个不同版本的 LLAVA 1.6（vicuna-7b 和 mistral-7b）。我在这里报告结果：
> 
> llava-mistral-7b 在 Science-QA 测试中不带 STIC-stage1 ![图像](https://private-user-images.githubusercontent.com/92557104/339676917-2e3f1d56-60b9-4108-b9af-9dbe710395be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1MzY2NzYsIm5iZiI6MTczMjUzNjM3NiwicGF0aCI6Ii85MjU1NzEwNC8zMzk2NzY5MTctMmUzZjFkNTYtNjBiOS00MTA4LWI5YWYtOWRiZTcxMDM5NWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI1VDEyMDYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2OGY0YzM3OTBjZTRlMGQ0MTljNTExNzBkNWUwMDEyZTdmMjIzOTJjNWQ5MDQ0OWFhMzE5NGM2NjljODZjZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1Xh69vgDhxMv_KqEbc5HiKDr1bom5AaNqHHvj-bb4qE)
> 
> llava-mistral-7b 在 Science-QA 测试中使用 STIC-stage1（使用您提供的 lora 权重） ![图像](https://private-user-images.githubusercontent.com/92557104/339676973-1bf08f33-2a29-45d2-8410-88d30fb80b83.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1MzY2NzYsIm5iZiI6MTczMjUzNjM3NiwicGF0aCI6Ii85MjU1NzEwNC8zMzk2NzY5NzMtMWJmMDhmMzMtMmEyOS00NWQyLTg0MTAtODhkMzBmYjgwYjgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI1VDEyMDYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiY2Y0N2VhN2E2Y2ZiYTc4NGExNGEwZjRmYTE2YzY2OTQ4OTI4ZWE4MDRjOWU4MGZmMWQxMzE5M2ZmMzFhZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.H8530o-PFjsRSQRx6xK8qM6JVguFxE75wVhqBj6CoeA)
> 
> 在这里不知何故，我们看到了结果，但它与论文不一致（大约 60）。
> 
> 我在 llava-vicuna-7b 中尝试了 STIC-stage1。在这里我没有看到任何改进。我们没有更改训练日期和设置。llava -vicuna-7b（原始） STIC-stage1 之后的 llava-vicuna-7b（在我们的环境中在 4 48G L20 上训练） ![图像](https://private-user-images.githubusercontent.com/92557104/339677757-0690ecdd-62a4-46b8-a310-8b0060abfa46.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1MzY2NzYsIm5iZiI6MTczMjUzNjM3NiwicGF0aCI6Ii85MjU1NzEwNC8zMzk2Nzc3NTctMDY5MGVjZGQtNjJhNC00NmI4LWEzMTAtOGIwMDYwYWJmYTQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI1VDEyMDYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2OTFhMjZmOWRhZDYzMjM4ZmU5ODExNmFjMjJlNGZhZGFmY2E3YjQxMzg3NDdhMmU5Yzg1MmY5MDE2NjUyNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.q0Ow-5vpppmpDtoAYnHceOUhdH4sCLbR4M-GECf33hw) ![图像](https://private-user-images.githubusercontent.com/92557104/339677857-34ab288b-d8c2-4476-9e21-aea78bb5c561.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1MzY2NzYsIm5iZiI6MTczMjUzNjM3NiwicGF0aCI6Ii85MjU1NzEwNC8zMzk2Nzc4NTctMzRhYjI4OGItZDhjMi00NDc2LTllMjEtYWVhNzhiYjVjNTYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI1VDEyMDYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwYzcxNTM1NTFjN2RhMTY1OGQwYzcwY2UwNWI0MjYzYTEyMjA4YmZiM2IzNDFlMTc0OWEyY2RjZGI3YWY5MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TBPku7RUub1Fb0fPr0CTu-_fw593KANHs6ssJzcGcBM)
> 
> 这里我也分享了训练损失日志和 lora 设置。它们看起来很正常
> 
> ![图像](https://private-user-images.githubusercontent.com/92557104/339678164-a223e813-27cc-4d32-bf44-080f0a217710.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1MzY2NzYsIm5iZiI6MTczMjUzNjM3NiwicGF0aCI6Ii85MjU1NzEwNC8zMzk2NzgxNjQtYTIyM2U4MTMtMjdjYy00ZDMyLWJmNDQtMDgwZjBhMjE3NzEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI1VDEyMDYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzNDBmY2M1YzU4ZWY2MmY5ZjAwZTZjODU3YzczNGJhNTk3ZmY4YzE2MDgwYzFiOGU0MDkyZjk2OTVmNDViMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JRFF9cgCKuE7uGCOteH8e_2veV9oyWARY0dEm7EHgOI)
> 
> 我该怎么做才能提高偏好优化？它真的帮助了我。谢谢

Hi, have you solved this?
